### PR TITLE
chore: merge dev into main

### LINKED ISSUE #21: Add edge case tests for autopilot policies
Enhance tests in projects/products/varina/backend/src/autopilot/autopilot_policy.rs to include edge cases and failure scenarios.

### LINKED ISSUE #23: Enhance code validation checks
Expand the check_common_issues method in projects/libraries/symbolic/src/validator/code_validator.rs to include checks for expect() and deprecated macros.

### LINKED ISSUE #26: Add realistic benchmarks for AST Core
Include benchmarks for realistic scenarios in projects/libraries/ast_core/benches/micro_bench.rs.

### LINKED ISSUE #27: Add security tests for Protocol Library
Write tests to validate security constraints in projects/libraries/protocol/README.md, such as payload size and allowed characters.

### LINKED ISSUE #104: Add edge case tests for common_json library
Add edge case tests for the common_json library to ensure robustness and handle unexpected inputs effectively.

### LINKED ISSUE #107: Add unit tests for symbolic_solver and code_validator
Add unit tests for symbolic_solver and code_validator to ensure edge cases are covered and improve robustness.

### LINKED ISSUE #111: Add benchmarks for ast_core and symbolic libraries
Add benchmarks for ast_core and symbolic libraries to identify performance bottlenecks.

### LINKED ISSUE #128: The panic message is in French ("Erreur lors de la visite de u64"). Error messages and panic messages should be in English to maintain consistency with the rest of the codebase.
The panic message is in French ("Erreur lors de la visite de u64"). Error messages and panic messages should be in English to maintain consistency with the rest of the codebase.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025383_

### LINKED ISSUE #129: The panic messages and variable name are in French ("Valeur valide", "Erreur lors de la visite de f64"). Error messages and identifiers should be in English to maintain consistency with the rest of the codebase.
The panic messages and variable name are in French ("Valeur valide", "Erreur lors de la visite de f64"). Error messages and identifiers should be in English to maintain consistency with the rest of the codebase.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025386_

### LINKED ISSUE #130: Multiple panic messages are in French ("Erreur lors de l'accès au champ", "Erreur lors de l'accès à l'index", "Erreur lors de l'accès au chemin", "Erreur lors de la définition du champ", "Erreur lors de la suppression du champ"). All error messages should be in English to maintain consistency with the rest of the codebase.
Multiple panic messages are in French ("Erreur lors de l'accès au champ", "Erreur lors de l'accès à l'index", "Erreur lors de l'accès au chemin", "Erreur lors de la définition du champ", "Erreur lors de la suppression du champ"). All error messages should be in English to maintain consistency with the rest of the codebase.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025390_

### LINKED ISSUE #131: The regex patterns for detecting expect and try! are being compiled on every call to check_common_issues. This is inefficient and will impact performance, especially when validating multiple code snippets. Consider compiling these regexes once at struct creation time or using lazy_static/once_cell to cache them. For example, store the compiled Regex as a field in CodeValidator or use a static lazy pattern.
The regex patterns for detecting expect and try! are being compiled on every call to check_common_issues. This is inefficient and will impact performance, especially when validating multiple code snippets. Consider compiling these regexes once at struct creation time or using lazy_static/once_cell to cache them. For example, store the compiled Regex as a field in CodeValidator or use a static lazy pattern.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025396_

### LINKED ISSUE #132: The println! debugging statements should be removed from test code. Tests should use assertions or testing frameworks instead of printing to stdout. These debug statements appear to be leftover from development and don't add value to the test suite.
The println! debugging statements should be removed from test code. Tests should use assertions or testing frameworks instead of printing to stdout. These debug statements appear to be leftover from development and don't add value to the test suite.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025398_

### LINKED ISSUE #133: The panic message is in French ("Erreur lors de la visite de i64"). Error messages and panic messages should be in English to maintain consistency with the rest of the codebase.
The panic message is in French ("Erreur lors de la visite de i64"). Error messages and panic messages should be in English to maintain consistency with the rest of the codebase.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025401_

### LINKED ISSUE #134: Multiple panic messages are in French ("Erreur lors de la définition du champ", "Erreur lors de l'accès au champ"). Error messages should be in English to maintain consistency with the rest of the codebase.
Multiple panic messages are in French ("Erreur lors de la définition du champ", "Erreur lors de l'accès au champ"). Error messages should be in English to maintain consistency with the rest of the codebase.
```suggestion
        panic!("Error while setting field: {:?}", e);
    }

    match json.get_field("key") {
        Ok(value) => assert_eq!(value, &Json::String("value".to_string())),
        Err(e) => panic!("Error while accessing field: {:?}", e),
```

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025404_

### LINKED ISSUE #135: Multiple panic messages are in French ("Erreur lors de la définition du champ", "Erreur lors de l'accès au champ", "Erreur lors de la suppression du champ"). Error messages should be in English to maintain consistency with the rest of the codebase.
Multiple panic messages are in French ("Erreur lors de la définition du champ", "Erreur lors de l'accès au champ", "Erreur lors de la suppression du champ"). Error messages should be in English to maintain consistency with the rest of the codebase.
```suggestion
        panic!("Error while setting field: {:?}", e);
    }
    match json.get_field("key") {
        Ok(value) => assert_eq!(value, &Json::String("value".to_string())),
        Err(e) => panic!("Error while accessing field: {:?}", e),
    }

    let mut json = Json::Object(JsonObject::new());
    if let Err(e) = json.set_field("key", Json::String("value".to_string())) {
        panic!("Error while setting field: {:?}", e);
    }
    if let Err(e) = json.remove_field("key") {
        panic!("Error while removing field: {:?}", e);
```

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025405_

### LINKED ISSUE #136: The panic message is in French ("Erreur lors de la conversion en JSON"). Error messages should be in English to maintain consistency with the rest of the codebase.
The panic message is in French ("Erreur lors de la conversion en JSON"). Error messages should be in English to maintain consistency with the rest of the codebase.
```suggestion
        Err(e) => panic!("Error while converting to JSON: {:?}", e),
```

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025391_

### LINKED ISSUE #137: The comment is in French ("Utilisation d'un type valide"). Comments should be in English to maintain consistency with the rest of the codebase.
The comment is in French ("Utilisation d'un type valide"). Comments should be in English to maintain consistency with the rest of the codebase.
```suggestion
    let seed = PhantomData::<()>; // Use of a valid type
```

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025394_

### LINKED ISSUE #138: The panic message is in French ("Le résultat n'est pas un tableau JSON"). Error messages should be in English to maintain consistency with the rest of the codebase.
The panic message is in French ("Le résultat n'est pas un tableau JSON"). Error messages should be in English to maintain consistency with the rest of the codebase.
```suggestion
        panic!("The result is not a JSON array");
```

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025408_

### LINKED ISSUE #139: The comment and println! statements are in French ("Initialisation de `self.value`", "Résultat de next_key_seed"). Comments and debug output should be in English to maintain consistency with the rest of the codebase. Additionally, println! statements should not be left in test code - consider using a testing assertion library or removing them.
The comment and println! statements are in French ("Initialisation de `self.value`", "Résultat de next_key_seed"). Comments and debug output should be in English to maintain consistency with the rest of the codebase. Additionally, println! statements should not be left in test code - consider using a testing assertion library or removing them.

_Originally posted by @Copilot in https://github.com/organization-ai-projects/automation_project/pull/127#discussion_r2752025410_

File : projects/libraries/common_json/src/deserialization/tests/json_map_access.rs

### PR DESCRIPTION
**Pull request by remi-bezot**
## Summary
- Merge dev into main (unit tests, benchmarks, validation improvements)
- Covers common_json, symbolic, protocol, and code_agent_sandbox
- Includes common_json test message translations, removed debug prints, and cached validator regexes

## Testing
- cargo test --workspace
- cargo fmt --all -- --check

## Issues Closed
Closes #104
Closes #107
Closes #21
Closes #27
Closes #26
Closes #111
Closes #23
Closes #128
Closes #129
Closes #130
Closes #131
Closes #132
Closes #133
Closes #134
Closes #135
Closes #136
Closes #137
Closes #138
Closes #139

### CODE REVIEW

File: projects/libraries/common_json/src/tests/access.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+#[test]
+fn test_remove_field() {
+    let mut json = Json::Object(HashMap::from_iter(vec![(
+        "key".to_string(),
+        Json::from("value"),
+    )]));
+
+    match json.remove_field("key") {
+        Ok(value) => assert_eq!(value, Some(Json::from("value"))),
+        Err(e) => panic!("Erreur lors de la suppression du champ : {:?}", e),
```
[the remark this answers is not in the document]
Comment (remi-bezot) [reply projects/libraries/common_json/src/tests/access.rs]: Issue #130

File: projects/libraries/symbolic/src/validator/code_validator.rs [hunk 2]
```diff
@@ -125,6 +127,19 @@ impl CodeValidator {
...
+        if expect_re.is_match(code) {
+            warnings
+                .push("Code contains expect calls (consider proper error handling)".to_string());
+        }
+
+        // Check for deprecated macros
+        let try_re = Regex::new(r"\btry\s*!\s*\(").expect("valid try macro regex");
+        if try_re.is_match(code) {
+            warnings.push("Code contains deprecated try! macro (use ? instead)".to_string());
+        }
```
[the remark this answers is not in the document]
Comment (remi-bezot) [reply projects/libraries/symbolic/src/validator/code_validator.rs]: Issue #131

File: projects/libraries/common_json/src/deserialization/tests/json_map_access.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_json_map_access_empty() {
+    let data: HashMap<String, Json> = HashMap::new();
+    let mut map = JsonMapAccess::new(data.iter());
+    let seed = PhantomData::<String>;
+    let result = map.next_key_seed(seed);
+    println!("Résultat de next_key_seed: {:?}", result);
```
[the remark this answers is not in the document]
Comment (remi-bezot) [reply projects/libraries/common_json/src/deserialization/tests/json_map_access.rs]: Issue #132

File: projects/libraries/common_json/src/tests/json.rs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+#[test]
+fn test_to_json_string_pretty() {
+    let value = vec!["test", "json"];
+    let result = to_json_string_pretty(&value);
+    match result {
+        Ok(json) => {
+            assert!(json.contains("test"));
+            assert!(json.contains("json"));
+        }
+        Err(e) => panic!("Erreur lors de la conversion en JSON : {:?}", e),
```
[the remark this answers is not in the document]
Comment (remi-bezot) [reply projects/libraries/common_json/src/tests/json.rs]: Issue #135

File: projects/libraries/common_json/src/tests/json_array_builder.rs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    let mut builder = JsonArrayBuilder::new();
+    builder = builder.element("value1");
+    builder = builder.element("value2");
+    let array = builder.build();
+    if let Json::Array(arr) = array {
+        assert_eq!(arr.len(), 2);
+        assert!(arr.contains(&Json::from("value1")));
+        assert!(arr.contains(&Json::from("value2")));
+    } else {
+        panic!("Le résultat n'est pas un tableau JSON");
```
[the remark this answers is not in the document]
Comment (remi-bezot) [reply projects/libraries/common_json/src/tests/json_array_builder.rs]: Issue #138

File: projects/libraries/common_json/src/deserialization/tests/json_map_access.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (remi-bezot) [reply projects/libraries/common_json/src/deserialization/tests/json_map_access.rs]: Issue #139
